bundles/org.openhab.binding.adorne/src/main/java/org/openhab/binding/adorne/internal/handler/AdorneHubHandler.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.adorne.internal.handler;
+
+import static org.openhab.binding.adorne.internal.AdorneBindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.adorne.internal.configuration.AdorneHubConfiguration;
+import org.openhab.binding.adorne.internal.hub.AdorneHubChangeNotify;
+import org.openhab.binding.adorne.internal.hub.AdorneHubController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AdorneHubHandler} manages the state and status of the Adorne Hub's devices.
+ *
+ * @author Mark Theiding - Initial contribution
+ */
+@NonNullByDefault
+public class AdorneHubHandler extends BaseBridgeHandler implements AdorneHubChangeNotify {
+    private Logger logger = LoggerFactory.getLogger(AdorneHubHandler.class);
+    private Bridge bridge;
+    private @Nullable AdorneHubController adorneHubController = null;
+
+    public AdorneHubHandler(Bridge bridge) {
+        super(bridge);
+        this.bridge = bridge;
```

Comment:
No need to cache the bridge

Reply:
Good point. Done. Applied the forEach here as well.